### PR TITLE
Rebalance (and fix) burning of items counted by charges

### PR DESCRIPTION
**Pull request by Coolthulhu**
In master, items that are counted by charges still accumulate burn like other items. They are immediately removed when their burnt value is 3 times their (stacks') volume, but are not otherwise damaged. This doesn't have any ill effects because it only affects liquids, but it makes gauging fuel quality harder and is "less realistic".
Another bug in master is that gasoline and diesel don't burn, but cookoff due to bad logic.

In this PR I change burning liquids (and other count_by_charges comestibles) so that they burn gradually, depending on fire size and default stack size. Those items do not accumulate `burnt` value at all.

I changed the hardcoded alcohol checking (for gin, vodka, brandy etc.) into check for alcohol material. Weak alcoholic drinks are handled by giving them alcohol and water materials and then checking for the alcohol being the only material (a bit of a placeholder here).

I fixed the gasoline cookoff bug.
I fixed generic (non special-cased) count_by_charge items burning too slow when in big stacks.

I adjusted two values: smoke value for alcohol (down from 5 to 1) and powder burns 100 times more intensely (older values were incredibly low).

Flour gets a `"powder"` material, so that it burns faster.

EDIT: Changed gasoline and diesel check into check for new material (named it `hydrocarbons`). This new material has fire resistance 0, meaning it is once again usable by internal furnace.

### CODE REVIEW

**Comment by Barhandar** [comment]
Does this allow gasoline to fuel the Internal Furnace again?

For that matter, how does IF even interact with charged items?

**Comment by Coolthulhu** [comment]
Forgot about gasoline furnace, fix'd in next commit.

**Comment by kevingranade** [comment]
Just FYI, we'd like to get rid of item 'charges' entirely at some point.

**Comment by Barhandar** [comment]
>Forgot about gasoline furnace, fix'd in next commit.

Yay! How does furnace even interact with it, though? E.g. I have 200 charges of gasoline (1 unit), I eat them, how much bionic power I get? What if 100 charges? 500?

**Comment by Coolthulhu** [comment]
> Just FYI, we'd like to get rid of item 'charges' entirely at some point.

Well, it still makes it act more "realistically" (and more like if it was all charge-less) by burning items in small batches rather than in one go.

How will lack of charges work with items with lots of charges, like gasoline?

>Yay! How does furnace even interact with it, though? E.g. I have 200 charges of gasoline (1 unit), I eat them, how much bionic power I get? What if 100 charges? 500?

Like it did before, I didn't rebalance it.
Apparently it gives `(volume + weight)/9` charges, halved for wood, divided by 4 for leather. Weight is in grams as far as I recall.

**Comment by KA101** [comment]
Hmm.  I don't think sugar should burn *quite* so fast (8,000 charges dropped on a lit fireplace, and gone in one turn) but wev.

File: src/field.cpp [hunk 1]
```diff
@@ -719,12 +716,13 @@ bool map::process_fields_in_submap( submap *const current_submap,
...
+                                // burn_amt will get multiplied by stack size in item::burn
                                 burn_amt = cur->getFieldDensity();
 
                             } else if( fuel->made_of("powder") ) {
-                                // Any powder will fuel the fire as much as its volume
+                                // Any powder will fuel the fire as 100 times much as its volume
                                 // but be immediately destroyed.
-                                time_added = vol;
+                                time_added = vol * 100;
                                 destroyed = true;
```
Comment (KA101) [inline src/field.cpp]: Yeah, OK, so dropping a ludicrous amount of powder on a fireplace means it keeps going way longer than expected.  So noted.

**Pull request merged**